sdk/go/pulumi/types.go

```diff
@@ -728,6 +746,11 @@ func awaitInputs(ctx context.Context, v, resolved reflect.Value) (bool, bool, []
 
 	contract.Assertf(valueType.AssignableTo(resolved.Type()), "%s not assignable to %s", valueType.String(), resolved.Type().String())
 
+	if v.CanInterface() && v.Type().Implements(resourceType) {
```

Comment:
Do we need the CanInterface check here? Isn't that covered by Implements?

Reply:
It is not, no--`CanInterface` just indicates whether or not the `Interface()` method of `reflect.Value` will panic.

Reply:
Ah, though we don't need the check b/c we're not _calling_ `Interface()`.